core/src/saros/util/PathUtils.java

```diff
@@ -0,0 +1,141 @@
+
+  /** The separator character used for portable paths. */
+  public static final String PORTABLE_SEPARATOR = "/";
+
+  /** The default separator character used on the Windows platform. */
+  private static final String WINDOWS_SEPARATOR = "\\";
+
+  /**
+   * Returns whether the given path is empty.
+   *
+   * @param path the path to check
+   * @return whether the given path is empty
+   */
+  public static boolean isEmpty(Path path) {
+    Objects.requireNonNull(path, "The given path must not be null");
+
+    Path normalized = path.normalize();
+
+    return normalized.getNameCount() == 1 && normalized.toString().equals("");
+  }
+
+  /**
+   * Removes the given number of segments from the start of the given path.
+   *
+   * <p>Returns an empty path if the number of segments to drop is larger or equal to the number of
+   * segments in the given path.
+   *
+   * @param path the path whose segments to drop
+   * @param count the number of segments to drop
+   * @return the remaining path
+   * @see Path#subpath(int, int)
+   */
+  public static Path removeFirstSegments(Path path, int count) {
+    Objects.requireNonNull(path, "The given path must not be null");
+
+    if (path.getNameCount() <= count) {
+      return Paths.get("");
+    }
+
+    return path.subpath(count, path.getNameCount());
+  }
+
+  /**
+   * Removes the given number of segments from the end of the given path.
+   *
+   * <p>Returns an empty path if the number of segments to drop is larger or equal to the number of
+   * segments in the given path.
+   *
+   * @param path the path whose segments to drop
+   * @param count the number of segments to drop
+   * @return the remaining path
+   * @see Path#subpath(int, int)
+   */
+  public static Path removeLastSegments(Path path, int count) {
+    Objects.requireNonNull(path, "The given path must not be null");
+
+    if (path.getNameCount() <= count) {
+      return Paths.get("");
+    }
+
+    return path.subpath(0, path.getNameCount() - count);
+  }
+
+  /**
+   * Returns a platform-independent string representation of the given path. The given path must not
+   * be absolute.
+   *
+   * <p>The string can be converted into a path for the local filesystem using {@link
+   * PathUtils#fromPortableString(String)}.
+   *
+   * @param path the path to convert
+   * @return a platform-independent string representation of the given path
+   * @see PathUtils#fromPortableString(String)
+   * @throws IllegalArgumentException if the given path is absolute
+   */
+  public static String toPortableString(Path path) {
+    Objects.requireNonNull(path, "The given path must not be null");
+
+    if (path.isAbsolute()) {
+      throw new IllegalArgumentException("The given path must not be absolute");
+    }
+
+    String localPathString = path.toString();
+
+    if (File.separator.equals(PORTABLE_SEPARATOR)) {
+      if (localPathString.contains(WINDOWS_SEPARATOR)) {
```

Comment:
Yeah, but I didn't want to overdo it with the special case handling. This can still be added later on if it seem important. 

Additionally, any of the forbidden characters should lead to an exception when trying to create a path containing them (at least I would hope so). This case, on the other hand, just leads to differing paths on the receiving side, which is much harder to notice/diagnose.